themes/docs-new/layouts/shortcodes/unified_mode.md

```diff
@@ -0,0 +1,192 @@
+
+- Declare a resource first and then declare actions on it
+- Write resources in run-time order
+
+### Resources with Changes to Internal Sub-resources
+
+Some custom resources are designed to create and edit other sub-resources as part of the resource declaration. In unified mode, the Chef Infra Client parses a resource code block that creates or edits a sub-resource and immediately and tries to apply that change, even though the sub-resource does not yet exist. This results in the execution of an incomplete resource.
+
+For example, with unified mode enabled, this code from the _dhcp cookbook_ is designed to create and edit a shared `dhcp_subnet` resource does not work as expected:
+
+```ruby
+# 'edit_resource' results in an incomple subresource
+sr = edit_resource(:dhcp_subnet, "#{new_resource.name}_sharedsubnet_#{subnet}") do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+end
+
+properties.each do |property, value|
+  sr.send(property, value)
+end
+```
+
+To correct custom resources that change sub-resources during their declaration, you can:
+
+- Apply properties in the code block (preferred)
+- Run the resource explicitly (not preferred)
+
+#### Apply Properties in the Code Block
+
+This pattern declares the sub-resource in one code block and then changes it in the next code block. This is the preferred pattern in unified mode because all resources execute in order at compile time.
+
+```ruby
+dhcp_subnet "#{new_resource.name}_sharedsubnet_#{subnet}" do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+
+  properties.each do |property, value|
+    send(property, value)
+  end
+end
+```
+
+#### Run the Resource Explicitly
+
+Another solution is to continue saving the resource as a variable, declaring `action :nothing` within the codeblock and then explicitly running the action in another codeblock.
+
+The pattern of saving a resource as a variable and then forcing it to run at compile time with an explicit `run_action` works as it has in the past, but it not a preferred pattern. Unified mode forces resource execution to compile time by default, which makes this pattern redundant.
+
+```ruby
+sr = edit_resource(:dhcp_subnet, "#{new_resource.name}_sharedsubnet_#{subnet}") do
+  owner new_resource.owner
+  group new_resource.group
+  mode new_resource.mode
+
+  ip_version new_resource.ip_version
+  conf_dir new_resource.conf_dir
+  shared_network true
+
+  action :nothing
+end
+
+properties.each do |property, value|
+  sr.send(property, value)
+end
+
+# Run the action explicitly
+sr.run_action(:create)
+```
+
+## Actions on Later Resources
+
+Since unified mode executes your resource as it is compiled, the behavior of immediate notifications to resources declared later is different.
```

Comment:
That feels clunky too.

```suggestion
Since unified mode executes your resource as it is compiled, `:immediate` notifications that execute later resources are handled differently.
```

I don't know if that's a very good suggestion, but it might be better 🤷